packages/react-bindings/src/hooks/client.ts

```diff
@@ -57,9 +57,11 @@ export const useCreateStreamVideoClient = ({
             console.error(`Failed to establish connection`, err);
           });
       }
-      return client.connectUser(user, tokenOrProvider).catch((err) => {
-        console.error(`Failed to establish connection`, err);
-      });
+      if (user.id && tokenOrProvider) {
+        return client.connectUser(user, tokenOrProvider).catch((err) => {
+          console.error(`Failed to establish connection`, err);
+        });
+      }
```

Comment:
Apologies for jumping in this PR but, why are these checks necessary? `user`, `user.id` and `tokenOrProvider` are marked as required.
If missing at runtime, we should throw an error instead of silently doing nothing.

Reply:
Yes, but does it make sense to call connectUser unnecessarily, if the user hasn't provided the token and user details appropriately. `user` is still fine, but there are cases where token might not be generated and therefore not be passed at all. In such a case, connectUser shouldn't be called from what I understand.

Reply:
Please correct me if I am wrong.